AirCasting/CoreData/Session.swift

```diff
@@ -31,69 +69,67 @@ public struct SessionUUID: Codable, RawRepresentable, Hashable, CustomStringConv
-        get { (value(forKey: "status") as? Int).flatMap(SessionStatus.init(rawValue:)) }
-        set { setValue(newValue?.rawValue, forKey: "status") }
+    public init(from decoder: Decoder) throws {
+        let singleValue = try decoder.singleValueContainer()
+        let rawValue = try singleValue.decode(String.self)
+        self.init(rawValue: rawValue)
     }
 
-    public var uuid: SessionUUID! {
-        get { SessionUUID(rawValue: value(forKey: "uuid") as! String) }
-        set { setValue(newValue.rawValue, forKey: "uuid") }
+    public func encode(to encoder: Encoder) throws {
+        var container = encoder.singleValueContainer()
+        try container.encode(description)
     }
 
-    public var deviceType: DeviceType? {
-        get { (value(forKey: "deviceType") as? Int).flatMap(DeviceType.init(rawValue:)) }
-        set { setValue(newValue?.rawValue, forKey: "deviceType") }
+    public var rawValue: String {
+        switch self {
+        case .mobile: return "MobileSession"
+        case .fixed: return "FixedSession"
+        case .unknown(let rawValue): return rawValue
+        }
     }
 
-    public var type: SessionType! {
-        get { SessionType(rawValue:(value(forKey: "type") as! String)) }
-        set { setValue(newValue.rawValue, forKey: "type") }
+    public init(rawValue: String) {
+        switch rawValue {
+        case "MobileSession": self = .mobile
+        case "FixedSession": self = .fixed
+        default: self = .unknown(rawValue)
+        }
     }
-}
 
-extension NSFetchRequest where ResultType == Session {
-    public func typePredicate(_ type: SessionType) -> NSPredicate {
-        NSPredicate(format: "type == \"\(type.rawValue)\"")
+    public var description: String {
+        switch self {
+        case .mobile: return NSLocalizedString("Mobile", comment: "Mobile user readable localized description")
+        case .fixed: return NSLocalizedString("Fixed", comment: "Fixed user readable localized description")
+        case .unknown: return NSLocalizedString("Other", comment: "Unknown user readable localized description")
+        }
     }
 }
-// MARK: Generated accessors for measurementStreams
-extension Session {
 
-    @objc(addMeasurementStreamsObject:)
-    @NSManaged public func addToMeasurementStreams(_ value: MeasurementStream)
-
-    @objc(removeMeasurementStreamsObject:)
-    @NSManaged public func removeFromMeasurementStreams(_ value: MeasurementStream)
+public enum SessionStatus: Int {
+    case NEW = -1
+    case RECORDING = 0
+    case FINISHED = 1
+    case DISCONNETCED = 2
+}
 
-    @objc(addMeasurementStreams:)
-    @NSManaged public func addToMeasurementStreams(_ values: NSSet)
+enum StreamingMethod: Int {
+    case CELLULAR = 0
+    case WIFI = 1
+}
 
-    @objc(removeMeasurementStreams:)
-    @NSManaged public func removeFromMeasurementStreams(_ values: NSSet)
+public enum DeviceType: Int, CustomStringConvertible {
+    case MIC = 0
+    case AIRBEAM3 = 1
```

Comment:
Well, in the Android app its value is probably 3 because they connect with all airbeams: AirBeam1, AirBeam2, and AirBeam3. In our case, we connect only with AB3 so I guess it can be 1. 
Unless we share this information with the web, do we?

Reply:
Okay, right, we don't :)